src/engine/board.ts

```diff
@@ -40,6 +40,16 @@ export default class Board {
         }
     }
 
+    public isPieceAtSquare(square: Square) {
+        let pieceAtSquare : Piece | undefined;
+        pieceAtSquare = this.getPiece(square);
```

Comment:
```suggestion
        const pieceAtSquare = this.getPiece(square);
```
we can cimplify these two lines to this, we don't need to add a type here as it is typed implicity by the return type of `getPiece`